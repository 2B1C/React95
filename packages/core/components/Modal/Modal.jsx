import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from '@xstyled/styled-components';
import { th } from '@xstyled/system';
import Draggable from 'react-draggable';

import Btn from '../shared-style/Btn';
import Button from '../Button';
import Icon from '../Icon';

const ModalWrapper = styled.div`
  display: flex;
  flex-direction: column;

  position: fixed;

  padding: 2 2 8;
  ${({ width, height }) => `width: ${width}px; height: ${height}px;`}

  top: 50px;

  background-color: bg;

  box-shadow: inset 1px 1px 0px 1px ${th('colors.white')},
    inset 0 0 0 1px ${th('colors.grays.3')}, 1px 1px 0 1px ${th('colors.black')};
`;

const TitleBar = styled.div`
  height: 18px;
  margin-bottom: 2;

  background-color: primary;
  color: ${th('colors.white')};
  padding: 2 2 0;

  display: flex;
`;

const Title = styled.div`
  flex-grow: 1;
  font-weight: bold;
`;

const OptionsBox = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;

  display: flex;
`;

const Option = styled(Btn)`
  margin-right: 2;
  padding: 0;

  width: 17px;
  height: 14px;
  min-width: 0;

  font-size: 10;

  &:last-child {
    margin-right: 0;
  }

  &:active {
    padding: 1 0 0 1;

    outline: none;
  }

  &:focus {
    box-shadow: inset 1px 1px 0px 1px ${th('colors.white')},
      inset -1px -1px 0px 1px ${th('colors.grays.3')};
  }
`;

Option.displayName = 'Option';

const Content = styled.div`
  flex-grow: 1;
  display: flex;
  flex-direction: column;

  padding: 6;
`;

const ButtonWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: ${props => props.buttonsAlignment};

  padding: 0 6 6 6;

  & ${Btn} {
    margin-right: 6;
    min-width: 70px;

    &:last-child {
      margin-right: 0;
    }
  }
`;

const MenuWrapper = styled.ul`
  display: flex;
  flex-direction: row;

  list-style: none;
  margin: 0;
  padding-left: 0;
  padding-bottom: 3;

  border-bottom-style: solid;
  border-width: 1;
  border-bottom-color: grays.3;

  box-shadow: 0 1px 0 0 ${th('colors.grays.0')};
`;

const MenuItem = styled.li`
  position: relative;
  padding-left: 6;
  padding-right: 6;

  user-select: none;

  ul {
    position: absolute;
    left: 0;
    color: ${th('colors.black')};
  }

  ${({ active }) =>
    active &&
    css`
      background-color: primary;
      color: ${th('colors.white')};
    `};
`;

MenuItem.displayName = 'MenuItem';

const Modal = ({
  closeModal,
  title,
  children,
  buttons,
  icon,
  menu,
  buttonsAlignment,
  defaultPosition,
  width,
  height,
  ...rest
}) => {
  const [menuOpened, setMenuOpened] = useState('');

  const iconStyle = {
    width: 15,
    height: 13,
    style: {
      marginRight: '4px',
    },
  };

  return (
    <>
      <Draggable handle=".draggable" defaultPosition={defaultPosition}>
        <ModalWrapper width={width} height={height} {...rest}>
          <TitleBar className="draggable">
            {icon && <Icon name={icon} {...iconStyle} />}
            <Title>{title}</Title>
            <OptionsBox>
              <Option>?</Option>
              <Option onClick={closeModal}>x</Option>
            </OptionsBox>
          </TitleBar>

          {menu.length > 0 && (
            <MenuWrapper>
              {menu.map(({ name, list }) => {
                const active = menuOpened === name;
                return (
                  <MenuItem
                    key={name}
                    onMouseDown={() => setMenuOpened(name)}
                    active={active}
                  >
                    {name}
                    {active && list}
                  </MenuItem>
                );
              })}
            </MenuWrapper>
          )}

          <Content onClick={() => setMenuOpened('')}>{children}</Content>
          {buttons.length > 0 && (
            <ButtonWrapper buttonsAlignment={buttonsAlignment}>
              {buttons.map((button, index) => (
                <Button
                  // eslint-disable-next-line react/no-array-index-key
                  key={index}
                  onClick={button.onClick}
                  value={button.value}
                />
              ))}
            </ButtonWrapper>
          )}
        </ModalWrapper>
      </Draggable>
    </>
  );
};

Modal.displayName = 'Modal';

Modal.propTypes = {
  icon: PropTypes.string,
  closeModal: PropTypes.func,
  title: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  buttons: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string.isRequired,
      onClick: PropTypes.func,
    }),
  ),
  menu: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      list: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
      ]).isRequired,
    }),
  ),
  buttonsAlignment: PropTypes.oneOf(['center', 'flex-start', 'flex-end']),
  defaultPosition: PropTypes.shape({
    x: PropTypes.number.isRequired,
    y: PropTypes.number.isRequired,
  }),
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

Modal.defaultProps = {
  icon: '',
  title: 'Modal',
  buttonsAlignment: 'flex-end',
  children: null,
  defaultPosition: { x: 0, y: 0 },
  buttons: [],
  menu: [],
  width: 300,
  height: 200,
  closeModal: () => {},
};

export default Modal;
