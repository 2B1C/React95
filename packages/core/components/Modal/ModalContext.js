import { createContext } from 'react';

const ModalContext = createContext({ windows: [] });

export default ModalContext;
