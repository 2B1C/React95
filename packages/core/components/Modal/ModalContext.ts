/* eslint-disable no-unused-vars */
import { createContext } from 'react';

import { IconProps } from '../Icon/Icon';

export type Windows = {
  icon?: IconProps['name'];
  title: string;
};

export interface IModalContextProps {
  windows: Array<Windows>;
  addWindows(window: Windows): void;
  removeWindows(title: string): void;
  setActiveWindow(title: string): void;
  activeWindow?: string;
}

const ModalContext = createContext<IModalContextProps>({
  windows: [],
  addWindows: () => {},
  removeWindows: () => {},
  setActiveWindow: () => {},
  activeWindow: '',
});

export default ModalContext;
