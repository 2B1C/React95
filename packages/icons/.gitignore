src/react/
src/index.ts
src/iconDemos.ts
png/
icons.css
