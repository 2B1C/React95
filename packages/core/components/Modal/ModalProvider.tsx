import { nanoid } from 'nanoid';
import * as React from 'react';

import ModalContext, { Windows } from './ModalContext';

// eslint-disable-next-line @typescript-eslint/ban-types
export type ModalProviderProps = {};

type WindowStack = Record<string, Windows>;

type WindowAction =
  | {
      type: 'ADD_WINDOW';
      id: string;
      window: Windows;
    }
  | {
      type: 'UPDATE_WINDOW';
      id: string;
      window: Windows;
    }
  | {
      type: 'REMOVE_WINDOW';
      id: string;
    }
  | {
      type: 'REMOVE_BY_TITLE';
      title: string;
    };

const windowStackReducer: React.Reducer<WindowStack, WindowAction> = (
  state,
  action,
) => {
  const newWindows = { ...state };
  switch (action.type) {
    case 'ADD_WINDOW': {
      newWindows[action.id] = action.window;
      return newWindows;
    }
    case 'REMOVE_WINDOW': {
      delete newWindows[action.id];
      return newWindows;
    }
    case 'UPDATE_WINDOW': {
      newWindows[action.id] = action.window;
      return newWindows;
    }
    case 'REMOVE_BY_TITLE': {
      Object.entries(newWindows).forEach(([id, window]) => {
        if (window.title === action.title) {
          delete newWindows[id];
        }
      });
      return newWindows;
    }
    default:
      return state;
  }
};

const ModalProvider: React.FunctionComponent = ({ children }) => {
  const [windows, dispatch] = React.useReducer(windowStackReducer, {});
  const [activeWindow, setActiveWindow] = React.useState<string>();

  const addWindows = (window: Windows) => {
    const id = nanoid();
    dispatch({ type: 'ADD_WINDOW', id, window });
    return id;
  };
  // TODO: Remove this method in the next major release - updating by IDs is more sensible
  const removeWindows = (title: string) => {
    dispatch({ type: 'REMOVE_BY_TITLE', title });
  };
  const removeWindowById = (id: string) => {
    dispatch({ type: 'REMOVE_WINDOW', id });
  };
  const updateWindow = (id: string, window: Windows) => {
    dispatch({ type: 'UPDATE_WINDOW', id, window });
  };

  return (
    <ModalContext.Provider
      value={{
        windows,
        addWindows,
        removeWindows,
        removeWindowById,
        updateWindow,
        setActiveWindow,
        activeWindow,
      }}
    >
      {children}
    </ModalContext.Provider>
  );
};

export default ModalProvider;
