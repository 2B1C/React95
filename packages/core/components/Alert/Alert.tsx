import React from 'react';
import styled, { StyledComponent } from '@xstyled/styled-components';

import Modal, { ModalProps } from '../Modal/Modal';

import icons from '@react95/icons';

export enum DialogImages {
  error = icons.user_4,
  info = icons.user_5,
  question = icons.user_3,
  warning = icons.user_2,
}

export type DialogImageProps = { type?: keyof typeof DialogImages };

export type DialogProps = StyledComponent<'div', any, {}, never> & {
  Message: StyledComponent<'div', any, {}, never>;
  Image: StyledComponent<'div', any, DialogImageProps, never>;
};

const Message = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

const Image = styled.div<DialogImageProps>`
  background-repeat: no-repeat;
  background-size: 70%;
  height: 45px;
  width: 45px;
  background-position: center;
  margin-right: 10;

  ${({ type }) =>
    `background-image: url(${type ? DialogImages[type] : DialogImages.error});`}
`;

const Dialog: DialogProps = Object.assign(
  styled.div`
    display: flex;
    flex-direction: row;
  `,
  {
    Message,
    Image,
  },
);

export type AlertProps = Omit<
  ModalProps,
  'closeModal'
> & {} & DialogImageProps & {
    message: string;
    closeAlert: ModalProps['closeModal'];
  };

const Alert: React.FC<AlertProps> = ({
  type,
  message,
  closeAlert,
  ...rest
}) => (
  <Modal closeModal={closeAlert} height="120" {...rest}>
    <Dialog>
      <Dialog.Image type={type} />
      <Dialog.Message>{message}</Dialog.Message>
    </Dialog>
  </Modal>
);

Alert.displayName = 'Alert';

Alert.defaultProps = {
  type: 'error',
  buttons: [{ value: 'OK', onClick: () => {} }],
  closeAlert: () => {},
  buttonsAlignment: 'center',
  defaultPosition: {
    x: Math.floor(window.innerWidth / 2) - 150,
    y: Math.floor(window.innerHeight / 2) - 80,
  },
};

export default Alert;
