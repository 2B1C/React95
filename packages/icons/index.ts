import icons from './src';

export default icons;
