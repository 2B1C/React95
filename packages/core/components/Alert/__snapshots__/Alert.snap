// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> Snapshots should match snapshot with error type 1`] = `
.c12 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
}

.c12:disabled {
  color: #868a8e;
}

.c12:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c12:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 1px #000000;
  width: auto;
  height: 120px;
  z-index: 1;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  color: #ffffff;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #000e7a;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
}

.c3 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: 2px;
}

.c4:last-child {
  margin-right: 0;
}

.c6 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-size: 10px;
}

.c6:disabled {
  color: #868a8e;
}

.c6:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c6:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c6:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c6:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
}

.c7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 6px 6px 6px;
}

.c11 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c11 .c5:last-child {
  margin-right: 0;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 {
  background-repeat: no-repeat;
  background-size: 70%;
  height: 45px;
  width: 45px;
  background-position: center;
  margin-right: 10px;
  background-image: url(test-file-stub);
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div>
  <div
    class="c0 react-draggable"
    height="120"
    style="transform: translate(362px,304px);"
  >
    <div
      class="c1 draggable"
    >
      <div
        class="c2"
      >
        Error
      </div>
      <ul
        class="c3"
      >
        <li
          class="c4"
        >
          <button
            class="c5 c6"
          >
            ?
          </button>
        </li>
        <li
          class="c4"
        >
          <button
            class="c5 c6"
          >
            x
          </button>
        </li>
      </ul>
    </div>
    <div
      class="c7"
    >
      <div
        class="c8"
      >
        <div
          class="c9"
          type="error"
        />
        <div
          class="c10"
        >
          Error
        </div>
      </div>
    </div>
    <div
      class="c11"
    >
      <button
        class="c5 c12"
        value="OK"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;

exports[`<Alert /> Snapshots should match snapshot with info type 1`] = `
.c12 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
}

.c12:disabled {
  color: #868a8e;
}

.c12:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c12:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 1px #000000;
  width: auto;
  height: 120px;
  z-index: 1;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  color: #ffffff;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #000e7a;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
}

.c3 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: 2px;
}

.c4:last-child {
  margin-right: 0;
}

.c6 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-size: 10px;
}

.c6:disabled {
  color: #868a8e;
}

.c6:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c6:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c6:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c6:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
}

.c7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 6px 6px 6px;
}

.c11 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c11 .c5:last-child {
  margin-right: 0;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 {
  background-repeat: no-repeat;
  background-size: 70%;
  height: 45px;
  width: 45px;
  background-position: center;
  margin-right: 10px;
  background-image: url(test-file-stub);
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div>
  <div
    class="c0 react-draggable"
    height="120"
    style="transform: translate(362px,304px);"
  >
    <div
      class="c1 draggable"
    >
      <div
        class="c2"
      >
        Info
      </div>
      <ul
        class="c3"
      >
        <li
          class="c4"
        >
          <button
            class="c5 c6"
          >
            ?
          </button>
        </li>
        <li
          class="c4"
        >
          <button
            class="c5 c6"
          >
            x
          </button>
        </li>
      </ul>
    </div>
    <div
      class="c7"
    >
      <div
        class="c8"
      >
        <div
          class="c9"
          type="info"
        />
        <div
          class="c10"
        >
          Info
        </div>
      </div>
    </div>
    <div
      class="c11"
    >
      <button
        class="c5 c12"
        value="OK"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;

exports[`<Alert /> Snapshots should match snapshot with question type 1`] = `
.c12 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
}

.c12:disabled {
  color: #868a8e;
}

.c12:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c12:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 1px #000000;
  width: auto;
  height: 120px;
  z-index: 1;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  color: #ffffff;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #000e7a;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
}

.c3 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: 2px;
}

.c4:last-child {
  margin-right: 0;
}

.c6 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-size: 10px;
}

.c6:disabled {
  color: #868a8e;
}

.c6:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c6:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c6:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c6:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
}

.c7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 6px 6px 6px;
}

.c11 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c11 .c5:last-child {
  margin-right: 0;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 {
  background-repeat: no-repeat;
  background-size: 70%;
  height: 45px;
  width: 45px;
  background-position: center;
  margin-right: 10px;
  background-image: url(test-file-stub);
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div>
  <div
    class="c0 react-draggable"
    height="120"
    style="transform: translate(362px,304px);"
  >
    <div
      class="c1 draggable"
    >
      <div
        class="c2"
      >
        question
      </div>
      <ul
        class="c3"
      >
        <li
          class="c4"
        >
          <button
            class="c5 c6"
          >
            ?
          </button>
        </li>
        <li
          class="c4"
        >
          <button
            class="c5 c6"
          >
            x
          </button>
        </li>
      </ul>
    </div>
    <div
      class="c7"
    >
      <div
        class="c8"
      >
        <div
          class="c9"
          type="question"
        />
        <div
          class="c10"
        >
          question
        </div>
      </div>
    </div>
    <div
      class="c11"
    >
      <button
        class="c5 c12"
        value="OK"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;

exports[`<Alert /> Snapshots should match snapshot with warning type 1`] = `
.c12 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
}

.c12:disabled {
  color: #868a8e;
}

.c12:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c12:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 1px #000000;
  width: auto;
  height: 120px;
  z-index: 1;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  color: #ffffff;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #000e7a;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
}

.c3 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: 2px;
}

.c4:last-child {
  margin-right: 0;
}

.c6 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,1px 1px 0 0px #000000;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-size: 10px;
}

.c6:disabled {
  color: #868a8e;
}

.c6:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c6:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c6:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c6:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
}

.c7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 6px 6px 6px;
}

.c11 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c11 .c5:last-child {
  margin-right: 0;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 {
  background-repeat: no-repeat;
  background-size: 70%;
  height: 45px;
  width: 45px;
  background-position: center;
  margin-right: 10px;
  background-image: url(test-file-stub);
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div>
  <div
    class="c0 react-draggable"
    height="120"
    style="transform: translate(362px,304px);"
  >
    <div
      class="c1 draggable"
    >
      <div
        class="c2"
      >
        Warning
      </div>
      <ul
        class="c3"
      >
        <li
          class="c4"
        >
          <button
            class="c5 c6"
          >
            ?
          </button>
        </li>
        <li
          class="c4"
        >
          <button
            class="c5 c6"
          >
            x
          </button>
        </li>
      </ul>
    </div>
    <div
      class="c7"
    >
      <div
        class="c8"
      >
        <div
          class="c9"
          type="warning"
        />
        <div
          class="c10"
        >
          Warning
        </div>
      </div>
    </div>
    <div
      class="c11"
    >
      <button
        class="c5 c12"
        value="OK"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;
